src/main/java/org/wildfly/security/http/impl/ClientCertAuthenticationMechanism.java

```diff
@@ -95,4 +145,44 @@ public void evaluateRequest(HttpServerRequest request) throws HttpAuthentication
         }
     }
 
+    private boolean attemptReAuthentication(HttpServerRequest request) throws HttpAuthenticationException {
+        CachedIdentityAuthorizeCallback authorizeCallback = new CachedIdentityAuthorizeCallback(createIdentityCache(request), true);
+        try {
+            MechanismUtil.handleCallbacks(CLIENT_CERT_NAME, callbackHandler, authorizeCallback);
+        } catch (AuthenticationMechanismException e) {
+            throw e.toHttpAuthenticationException();
+        } catch (UnsupportedCallbackException e) {
+            throw log.mechCallbackHandlerFailedForUnknownReason(CLIENT_CERT_NAME, e).toHttpAuthenticationException();
+        }
+        if (authorizeCallback.isAuthorized()) {
+            return succeed(request);
+        }
+        return false;
+    }
+
+    private Function<SecurityDomain, IdentityCache> createIdentityCache(HttpServerRequest request) {
+        SSLSession sslSession = request.getSSLSession();
+        if (sslSession == null) {
+            return null;
+        }
+        return securityDomain -> new IdentityCache() {
+
+            final ConcurrentHashMap<SecurityDomain, CachedIdentity> identities = (ConcurrentHashMap<SecurityDomain, CachedIdentity>) SSLUtils.computeIfAbsent(sslSession, "org.wildfly.elytron.identity-cache", key -> new ConcurrentHashMap());
```

Comment:
Invalid usage of a raw type, which is probably why you had to add this cast too.